### PR TITLE
Apple m1

### PR DESCRIPTION
**Pull request by ghost**
#236

### CODE REVIEW

**Comment by BigslimVdub** [comment]
I will try to get to this today and let you know

**Comment by BigslimVdub** [comment]
Hmm build failed, Cxx not supporting -march=x86-64

you need to remove that out of line 443 of cmake lists.txt to match pr and appears to build then

File: CMakeLists.txt [hunk 1]
```diff
@@ -440,7 +440,7 @@ if (CMAKE_SYSTEM_NAME MATCHES "(SunOS|Solaris)")
 endif ()
 
 if (APPLE AND NOT IOS)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=x86-64 -fvisibility=default")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=default -std=c++11")
```
Comment (ghost) [inline CMakeLists.txt:443]: @BigslimVdub , isn't this what you mean?

**Comment by BigslimVdub** [comment]
Ya let me test again when I get a chance and will review 
Can you rebase your one commit?

**Comment by ghost** [comment]
That was already in the PR, since when you tested it previously. So I am surprised you got that error. It should be ready for testing again please try. I don't understand what this means: "Can you rebase your one commit?" rebase to what? I rebased to master as stoffu pushed a github action fix to prevent the failing windows build.

File: CMakeLists.txt (hunk 1 above)
Comment (BigslimVdub) [reply CMakeLists.txt:443]: ```set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=default")```

will set it straight to compile

Comment (ghost) [reply CMakeLists.txt:443]: What is the problem with `-std=c++11`?

Comment (BigslimVdub) [reply CMakeLists.txt:443]: ``-std=c++11`` does not build on M1 which this PR is intended for.

edit: disregard, mobile format was messed up and I didn't realize actual updated commit.

**Review by BigslimVdub — APPROVED** [review]
builds on M1 now

**Review by stoffu — APPROVED** [review]

**Pull request merged**